systemFlake.nix

```diff
@@ -26,14 +29,14 @@
 
 let
   inherit (flake-utils-plus.lib) eachSystem patchChannel;
-  inherit (builtins) foldl' mapAttrs removeAttrs attrValues;
+  inherit (builtins) foldl' mapAttrs removeAttrs attrValues isAttrs isList;
 
   # ensure for that all expected, but no extra attrs are present
-  validateHost =
-    { channelName ? null
-    , system ? null
-    , output ? null
-    , builder ? null
+  evalHostArgs =
+    { channelName ? "nixpkgs"
+    , system ? "x86_64-linux"
+    , output ? "nixosConfigurations"
+    , builder ? channels.${channelName}.input.lib.nixosSystem
```

Comment:
@blaggacao Would you be ok with this? Don't do any merging here, just validate and set defaults. If not I can just use the new merge function for folding hosts and merging outputs.

For this to work, we have to deprecate sharedModules and sharedExtraArgs. Which I think @gytis-ivaskevicius was ok with too. It makes the code simpler and cleans up api.

Reply:
This sound pretty good given the generalization you made out of the merge semantics.